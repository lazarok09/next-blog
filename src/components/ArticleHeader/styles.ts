import styled, { css } from "styled-components";
import { Title as HeadingStyles } from "../Heading/styles";

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
      @media ${theme.media.lteMedium} {
        font-size: calc(${theme.font.sizes.medium} + 1rem);
      }
    }
  `}
`;
export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: ${theme.font.sizes.medium};
    @media ${theme.media.lteMedium} {
      font-size: calc(${theme.font.sizes.small} + 0.2rem);
    }
  `}
`;
export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
