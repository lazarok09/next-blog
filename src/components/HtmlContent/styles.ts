import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    p {
      margin: ${theme.spacings.xlarge} 0;
    }
    a,
    a:visited,
    a:link {
      color: ${theme.colors.secondary};
      transition: all 300ms ease-in-out;
      text-decoration: none;
    }
    a:hover {
      filter: brightness(80%);
      text-decoration: underline;
    }
    pre, code{
      background: ${theme.colors.primary};
      padding: ${theme.spacings.medium};
      font-family: monospace;
      color: ${theme.colors.white};
      margin: ${theme.spacings.xlarge} 0;
      width: 100%;
      overflow-x: auto;
      font-size: ${theme.font.sizes.small};
    }
  }
    img {
        width: 100%;

    }
    .image {
      background: ${theme.colors.mediumGray};
      line-height: 0;
      margin: ${theme.spacings.xlarge} 0;
      max-width: 100%;

    }
    .image figcaption {
      font-size: ${theme.font.sizes.small};
      padding: ${theme.spacings.small};
      text-align: center;
      line-height: 1.3;
    }
    .image-style-side {
      float: right;
      max-width: 50%;
      margin: 0 ${theme.spacings.medium};
    }
    hr {
      border: none;
      border-bottom: 0.1rem solid ${theme.colors.mediumGray};
    }
    ul,
    ol {
      margin: ${theme.spacings.xlarge};
    }
    .table {
      max-width: 100%;
      overflow: hidden;
      overflow-x: auto;
    }
    table {
      width: 100%;
      border-collapse: collapse;
    }
    table th,
    table td{
      padding: ${theme.spacings.small};
      border: 0.1rem solid ${theme.colors.mediumGray};
    }
    }
    iframe {
    max-width: 100%;
    height: 50vh;
    }
    @media ${theme.media.lteMedium} {
      font-size: 2rem;

      .image-style-side {
        float: none;
        max-width: 100%;
        margin: 0;
      }
    }
  `}
`;
