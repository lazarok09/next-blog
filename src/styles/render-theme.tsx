import { render, RenderResult } from '@testing-library/react';

import { ThemeProvider } from 'styled-components';
import { BlogThemeProvider } from '../contexts/BlogThemeContext';

import { theme } from './theme';

export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<BlogThemeProvider>{children}</BlogThemeProvider>);
};
