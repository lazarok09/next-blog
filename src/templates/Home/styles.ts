import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    & > img {
      height: 40px;
    }
    background: ${theme.colors.white};
  `}
`;
