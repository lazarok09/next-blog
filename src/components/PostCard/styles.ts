import styled, { css } from "styled-components";
import { Title as HeadingStyles } from "../Heading/styles";
export const Wrapper = styled.div`
  ${({ theme }) => css`
    ${HeadingStyles} {
      margin: 0;
      margin-top: calc(${theme.spacings.small} - 0.6rem);
      margin-bottom: ${theme.spacings.small};
    }

    & a {
      text-decoration: none;
      color: inherit;
      transition: all 300ms ease-in-out;
    }
    & a:hover {
      color: ${theme.colors.secondary};
    }
    // when we hover this element & i want to change the behavier of the a tag
    &:hover a {
      color: ${theme.colors.secondary};
    }
    &:hover img {
      opacity: 0.8;
    }
  `}
`;
export const Cover = styled.img`
  ${() => css`
    max-width: 100%;
    height: 100%;
    max-height: 280px;
    transition: opacity 300ms ease-in-out;
  `}
`;
export const Excerpt = styled.p`
  ${() => css`
    word-break: break-word;
  `}
`;
