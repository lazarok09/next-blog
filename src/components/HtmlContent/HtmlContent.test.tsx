import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { HtmlContent } from '.';

describe('<TextComponent />', () => {
  it('should render TextComponent with Children as text', () => {
    renderTheme(<HtmlContent html={'Children'} />);
    expect(screen.getByText('Children')).toBeInTheDocument();
  });

  it('should render HtmlContent with correct font size', () => {
    renderTheme(<HtmlContent html={'Children'} />);
    const paragraph = screen.getByText('Children');
    expect(paragraph).toHaveStyle({
      'font-size': 'calc(1.6rem + 0.2rem)',
    });
  });
  it('should render HtmlContent with a heading html', () => {
    renderTheme(<HtmlContent html={'<h1>Hello</h1>'} />);
    const heading = screen.getByRole('heading', { name: 'Hello' });
    expect(heading).toBeInTheDocument();
  });
  it('should match snapshot', () => {
    const { container } = renderTheme(<HtmlContent html={'Children'} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
