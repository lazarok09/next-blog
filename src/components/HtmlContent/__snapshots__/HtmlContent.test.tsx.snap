// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextComponent /> should match snapshot 1`] = `
.c0 {
  font-size: calc(1.6rem + 0.2rem);
  line-height: 1.5;
}

.c0 p {
  margin: 2.4rem 0;
}

.c0 a,
.c0 a:visited,
.c0 a:link {
  color: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 a:hover {
  -webkit-filter: brightness(80%);
  filter: brightness(80%);
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 code {
  font-family: monospace;
  color: #dc143c;
  font-size: 1.6rem;
  background: #DDDDDD;
  padding: 0.2rem;
  margin: 0.2rem;
}

.c0 pre {
  background: #000000;
  padding: 2.4rem;
  font-family: monospace;
  color: #FFFFFF;
  margin: 4.0rem 0;
  margin: 2.4rem 0;
  width: 100%;
  overflow-x: auto;
  font-size: 1.6rem;
}

.c0 pre code {
  color: inherit;
  background: inherit;
}

.c0 img {
  width: 100%;
}

.c0 .image {
  background: #DDDDDD;
  line-height: 0;
  margin: 2.4rem 0;
  max-width: 100%;
}

.c0 .image figcaption {
  font-size: 1.6rem;
  padding: 1.6rem;
  text-align: center;
  line-height: 1.3;
}

.c0 .image-style-side {
  float: right;
  max-width: 50%;
  margin: 0 2.4rem;
}

.c0 hr {
  border: none;
  border-bottom: 0.1rem solid #DDDDDD;
}

.c0 ul,
.c0 ol {
  margin: 2.4rem 4.0rem;
}

.c0 .table {
  max-width: 100%;
  overflow: hidden;
  overflow-x: auto;
}

.c0 table {
  width: 100%;
  border-collapse: collapse;
  margin: 2.4rem 0;
}

.c0 table th,
.c0 table td {
  padding: 1.6rem;
  border: 0.1rem solid #DDDDDD;
}

.c0 blockquote {
  border-left: 0.5rem solid #dc143c;
  color: #AAAAAA;
  -webkit-filter: brightness(80%);
  filter: brightness(80%);
  padding-left: 2.4rem;
  font-style: italic;
  margin: 2.4rem;
}

.c0 iframe {
  max-width: 100%;
  height: 50vh;
}

@media (max-width:768px) {
  .c0 {
    font-size: 2rem;
  }

  .c0 .image-style-side {
    float: none;
    max-width: 100%;
    margin: 0;
  }
}

<div
  class="c0"
>
  Children
</div>
`;
